### PR TITLE
[VrtNU] fix empty title in json test

### LINKED ISSUE #2146: VrtNu download fails on some videos
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2021.12.27**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Belgium

### Description

Although most downloads on vrtnu (with credentials) still work, one series fails: (run from local git, but same output from ytdlp)

```
[debug] Command-line config: ['https://www.vrt.be/vrtnu/a-z/raymond-goethals--magie-in-marseille/1/raymond-goethals--magie-in-marseille-s1a1/', '-v', '--netrc']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.27 [6223f67a8] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: e540c56f3
[debug] Python version 3.9.7 (CPython 64bit) - Linux-5.13.0-22-generic-x86_64-with-glibc2.34
[debug] exe versions: ffmpeg 4.4 (setts), ffprobe 4.4, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, secretstorage, sqlite
[debug] Proxy map: {}
[VrtNU] Logging in
[VrtNU] Requesting XSRF Token
[VrtNU] Performing login
[debug] [VrtNU] Extracting URL: https://www.vrt.be/vrtnu/a-z/raymond-goethals--magie-in-marseille/1/raymond-goethals--magie-in-marseille-s1a1/
[VrtNU] raymond-goethals--magie-in-marseille-s1a1: Downloading webpage
[debug] [Canvas] Extracting URL: https://mediazone.vrt.be/api/v1/vrtvideo/assets/pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c
[Canvas] pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c: refreshtoken: Retrieve vrtnutoken
[Canvas] pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c: Downloading token
[Canvas] pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c: Downloading video JSON
ERROR: [Canvas] pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c: None
  File "/home/xyz/Development/yt-dlp/yt_dlp/extractor/common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "/home/xyz/Development/yt-dlp/yt_dlp/extractor/canvas.py", line 85, in _real_extract
    raise ExtractorError(data.get('message') or code, expected=True)
```

I debugged this issue, and apparently, this video has an empty `title` element in its json: 
`{'duration': 3121060, 'skinType': 'vod', 'title': '', 'shortDescription': None, 'drm': 'vrt|2021-12-28T.....Z|.../...', 'drmExpired': '2021-12-28...', 'aspectRatio': '16:9', 'posterImageUrl': None, 'channelId': None, 'targetUrls': [{'type': 'mpeg_dash', 'url': 'https://remix-cf.lwc.vrtcdn.be/remix/0603b9c1-b117-4caf-a4c1-xxx/remix.ism/.mpd'}, {'type': 'hls_aes', 'url': 'https://remix-cf.lwc.vrtcdn.be/remix/0603b9c1-b117-4caf-a4c1-xxx/remix_aes.ism/.m3u8'}, {'type': 'hls', 'url': 'https://remix-cf.lwc.vrtcdn.be/remix/0603b9c1-b117-4caf-a4c1-xx/remix.ism/.m3u8'}], 'playlist': {'content': [{'eventType': 'ONBOARDING', 'duration': 3000, 'skippable': False}, {'eventType': 'STANDARD', 'duration': 3121060, 'skippable': True}]}}`
 
The test on https://github.com/yt-dlp/yt-dlp/blob/aeec0e44e243dc625c8ff19964a624188d43e84b/yt_dlp/extractor/canvas.py#L79  seems to be the culprit. I think the test is not needed as issues with logon will be catched more early during the Token-logon steps. 

### Verbose log

```shell
[debug] Command-line config: ['https://www.vrt.be/vrtnu/a-z/raymond-goethals--magie-in-marseille/1/raymond-goethals--magie-in-marseille-s1a1/', '-v', '--netrc']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.27 [6223f67a8] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: e540c56f3
[debug] Python version 3.9.7 (CPython 64bit) - Linux-5.13.0-22-generic-x86_64-with-glibc2.34
[debug] exe versions: ffmpeg 4.4 (setts), ffprobe 4.4, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, secretstorage, sqlite
[debug] Proxy map: {}
[VrtNU] Logging in
[VrtNU] Requesting XSRF Token
[VrtNU] Performing login
[debug] [VrtNU] Extracting URL: https://www.vrt.be/vrtnu/a-z/raymond-goethals--magie-in-marseille/1/raymond-goethals--magie-in-marseille-s1a1/
[VrtNU] raymond-goethals--magie-in-marseille-s1a1: Downloading webpage
[debug] [Canvas] Extracting URL: https://mediazone.vrt.be/api/v1/vrtvideo/assets/pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c
[Canvas] pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c: refreshtoken: Retrieve vrtnutoken
[Canvas] pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c: Downloading token
[Canvas] pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c: Downloading video JSON
ERROR: [Canvas] pbs-pub-975eb3d7-86af-4a92-81cc-80074a58c8ff$vid-2602567b-2db5-4b8d-a533-51f24688bd5c: None
  File "/home/xyz/Development/yt-dlp/yt_dlp/extractor/common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "/home/xyz/Development/yt-dlp/yt_dlp/extractor/canvas.py", line 85, in _real_extract
    raise ExtractorError(data.get('message') or code, expected=True)
```

### PR DESCRIPTION
**Pull request by pgaig**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### fixes download issue where internal JSON has empty title

fixes #2146

internal json has _empty_ `title` >> adapt test to test if key exists:
`{'duration': 3121060, 'skinType': 'vod', 'title': '', 'shortDescription': None, 'drm': 'vrt|2021-12-28T.....Z|.../...', 'drmExpired': '2021-12-28...', 'aspectRatio': '16:9', 'posterImageUrl': None, 'channelId': None, 'targetUrls': [{'type': 'mpeg_dash', 'url': 'https://remix-cf.lwc.vrtcdn.be/remix/0603b9c1-b117-4caf-a4c1-xxx/remix.ism/.mpd'}, {'type': 'hls_aes', 'url': 'https://remix-cf.lwc.vrtcdn.be/remix/0603b9c1-b117-4caf-a4c1-xxx/remix_aes.ism/.m3u8'}, {'type': 'hls', 'url': 'https://remix-cf.lwc.vrtcdn.be/remix/0603b9c1-b117-4caf-a4c1-xx/remix.ism/.m3u8'}], 'playlist': {'content': [{'eventType': 'ONBOARDING', 'duration': 3000, 'skippable': False}, {'eventType': 'STANDARD', 'duration': 3121060, 'skippable': True}]}}`

### CODE REVIEW

**Comment by pukkandan** [comment]
The title returned by extractor should never be empty

**Comment by pgaig** [comment]
Hi @pukkandan , thanks for your feedback

If I understand the code correctly, this empty `title` field is returned by the vrtnu/canvas backend. 
https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/canvas.py#L73-L85

The check on `title` here only seems to be an error-check to identify if the logon / authorization was OK. 
I also tried to remove the complete `if`-block completely > this works:
https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/canvas.py#L79-L85

In other words the `if` is dead code as the actual logon is handled elsewhere: 
https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/canvas.py#L284-L315 

This PRQ tried to be as conservative a possible, because there are multiple URLs supported for that exractor. 

Please advise on how to tackle the issue.
